### PR TITLE
Restart server on failure when running Windows app

### PR DESCRIPTION
**Pull request by jmorganca**

### CODE REVIEW

File: app/lifecycle/server.go [hunk 1]
```diff
@@ -143,15 +147,12 @@ func SpawnServer(ctx context.Context, command string) (chan int, error) {
 			default:
 				crashCount++
 				slog.Warn(fmt.Sprintf("server crash %d - exit code %d - respawning", crashCount, code))
-				time.Sleep(500 * time.Millisecond)
-				if err := cmd.Start(); err != nil {
```
Comment (jmorganca) [inline app/lifecycle/server.go:147]: `Start()` can only be called once for each `exec.Cmd` instance and so we need to create a new one on every restart

File: app/lifecycle/server.go [hunk 2]
```diff
@@ -43,36 +43,21 @@ func getCLIFullPath(command string) string {
...
-
-	logDir := filepath.Dir(ServerLogFile)
-	_, err := os.Stat(logDir)
-	if errors.Is(err, os.ErrNotExist) {
-		if err := os.MkdirAll(logDir, 0o755); err != nil {
-			return done, fmt.Errorf("create ollama server log dir %s: %v", logDir, err)
-		}
-	}
-
+func start(ctx context.Context, command string) (*exec.Cmd, error) {
```
Comment (jmorganca) [inline app/lifecycle/server.go:46]: `start` factors out the setup of the `exec.Cmd`

File: app/lifecycle/server.go [hunk 3]
```diff
@@ -117,19 +102,38 @@ func SpawnServer(ctx context.Context, command string) (chan int, error) {
...
+	done := make(chan int)
+
 	go func() {
 		// Keep the server running unless we're shuttind down the app
 		crashCount := 0
 		for {
+			slog.Info("starting server...")
+			cmd, err := start(ctx, command)
+			if err != nil {
+				slog.Error(fmt.Sprintf("failed to start server %s", err))
```
Comment (dhiltgen) [inline app/lifecycle/server.go:133]: I think this should continue the for loop so we don't crash on a nil pointer with cmd.Wait() below.

**Review by dhiltgen — APPROVED** [review]
One glitch, but otherwise looks good.

File: app/lifecycle/server.go (hunk 3 above)
Comment (jmorganca) [reply app/lifecycle/server.go:133]: Ah great catch. Thank you!

**Pull request merged**